grammar.js

```diff
@@ -28,5 +28,41 @@ module.exports = grammar(C, {
       field('declarator', $._declarator),
       field('body', $.compound_statement),
     ),
+    
+    _field_declaration_list_item: $ => choice(
+      alias(choice($.type_field_declarator_macro, $.uint_field_declarator_macro), $.preproc_call_expression),
+      $.field_declaration,
+      $.preproc_def,
+      $.preproc_function_def,
+      $.preproc_call,
+      alias($.preproc_if_in_field_declaration_list, $.preproc_if),
+      alias($.preproc_ifdef_in_field_declaration_list, $.preproc_ifdef),
+    ),
+    
+    uint_field_declarator_macro: $ => choice(
+      seq(
+        field('macro_name', '__uint'),
+        '(',
+        $.identifier,
+        ',',
+        choice(prec(3, $.type_descriptor), prec(2, $.expression)), // TODO: $.expression is generic, choices needs to be more specific in future
```

Comment:
Nit: Style-wise, let's please honor some line length limit of 100 characters (or 80, as you seemingly did below) or so, max, and keep comments on the previous line.

```suggestion
        // TODO: $.expression is generic, choices needs to be more specific in future
        choice(prec(3, $.type_descriptor), prec(2, $.expression)),
```

Personally, I'd remove the comment altogether. I fail to see the issue here; we don't have to be overly prescriptive about what is being accepted. It seems that if anything `expression` is more flexible than need be (though I am not convinced it is?), which I'd think is fine. Or am I missing something?

That being said, still on this line, why do we need these `prec` invocations? Everything seems fine without.

Reply:
I'll remove the comment, I misunderstood your original comment about (*foo)() types being an invalid argument and was commenting `$.expression` would allow that. I'll remove prec here as I see your point, I did this to preemptively avoid conflicts but I see the parse doesn't warn of conflicts after I remove.